src/structs.py

```diff
@@ -1240,6 +1243,19 @@ class DemonstrationResponse(Response):
     teacher_traj: Optional[LowLevelTrajectory]
 
 
+class Monitor(abc.ABC):
```

Comment:
i think this should go in utils.py? it's really not a core struct, it's more of a helpful utility?